src/NServiceBus.Core/obsoletes.cs

```diff
@@ -347,6 +347,14 @@ public static void RegisterEncryptionService(this EndpointConfiguration config,
             throw new NotImplementedException();
         }
     }
+
+    [ObsoleteEx(
+        Message = "If self-hosting, call your start/stop code before/after `Endpoint.Create` or `Endpoint.Stop`. If using NServiceBus.Host or NServiceBus.Host.AzureCloudService, implement `IWantToRunWhenEndpointStartsAndStops` instead",
```

Comment:
I like that better. Wasn't sure how verbose we were allowed to be in these error messages. Slight change to the first sentence:

> IWantToRunWhenBusStartsAndStops has been moved to the host implementations.

"removed from the core" is implied.